sqlvm-alwayson-cluster/nested/configuringAlwaysOn.json

```diff
@@ -0,0 +1,130 @@
+{  
+   "$schema":"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+   "contentVersion":"1.0.0.0",
+   "parameters":{  
+      "sqlVMName":{  
+         "type":"string"
+      },
+      "location":{  
+         "type":"string"
+      },
+      "adminUsername":{  
+         "type":"string"
+      },
+      "adminPassword":{  
+         "type":"securestring"
+      },
+      "domainName":{  
+         "type":"string"
+      },
+      "sqlAOEPName":{  
+         "type":"string"
+      },
+      "sqlServerServiceAccountUserName":{  
+         "type":"string"
+      },
+      "sqlServerServiceAccountPassword":{  
+         "type":"securestring"
+      },
+      "createClusterModulesURL":{  
+         "type":"string"
+      },
+      "createClusterConfigurationFunction":{  
+         "type":"string"
+      },
+      "clusterName":{  
+         "type":"string"
+      },
+      "sharePath":{  
+         "type":"string"
+      },
+      "sqlAOAGName":{  
+         "type":"string"
+      },
+      "sqlAOListenerName":{  
+         "type":"string"
+      },
+      "sqlAOListenerPort":{  
+         "type":"string"
+      },
+      "sqlLBName":{  
+         "type":"string"
+      },
+      "sqlLBIPAddress":{  
+         "type":"string"
+      },
+      "adPDCVMName":{  
+         "type":"string"
+      },
+      "sqlwVMName":{  
+         "type":"string"
+      },
+      "numberOfDisks":{  
+         "type":"string"
+      },
+      "workloadType":{  
+         "type":"string"
+      }
+   },
+   "resources":[  
+      {  
+         "type":"Microsoft.Compute/virtualMachines/extensions",
+         "name":"[concat(parameters('sqlVMName'),'1/configuringAlwaysOn')]",
+         "apiVersion":"2015-06-15",
+         "location":"[parameters('location')]",
+         "properties":{  
+            "publisher":"Microsoft.Powershell",
+            "type":"DSC",
+            "typeHandlerVersion":"2.14",
+            "autoUpgradeMinorVersion":false,
```

Comment:
This template is also what we use in the Market place version which we only use the tested version and if there is a newer version we won't upgrade until the newer version is fully tested. Since we will maintain this template we can take care of the version problem.